File: app/src/main/java/cn/peterzhen/demo/MainActivity.java
package cn.peterzhen.demo;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;

import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import cn.cbsd.mvplibrary.mvp.XActivity;
import cn.cbsd.mvplibrary.widget.IosDialog;
import es.dmoral.toasty.MyToast;

public class MainActivity extends XActivity {

    @BindView(R.id.button1)
    Button mButton1;
    @BindView(R.id.button2)
    Button mButton2;
    @BindView(R.id.textView)
    TextView mTextView;
    @BindView(R.id.activity_main)
    LinearLayout mActivityMain;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void initData(Bundle savedInstanceState) {
        MyToast.init(getApplication(), true, true);

        mButton1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Router.newIntent(context)
                //        .to(FrameActivity.class)
                //        .launch();
                showDialog();
            }
        });

        mButton2.setOnClickListener(view -> {
            showDialog2();
            //Router.newIntent(context)
            //        .to(KotlinActivity.class)
            //        .launch();
        });

    }

    private void showDialog() {
        List<String> list = new ArrayList();
        for (int i = 0; i < 20; i++) {
            list.add("item"+(i + 1));
        }
        //new IosSheetDialog(context).builder()
        //        .addListItems(list,which -> {
        //            getvDelegate().showInfo(String.format("点击的是%d",which));
        //        })
        //        .show();
        new IosDialog(context).builder()
                .setItems(list,0,which -> {
                    getvDelegate().showInfo(String.format("点击的是%d",which));
                })
                .show();

    }

    private void showDialog2() {
        List<String> list = new ArrayList();
        for (int i = 0; i < 20; i++) {
            list.add("item"+(i + 1));
        }
        new MaterialDialog.Builder(context)
                .items(list)
                .itemsCallbackSingleChoice(0, (dialog, itemView, which, text) -> {
                    getvDelegate().showInfo(String.format("点击的是%d",which));
                    return false;
                })
                .show();
    }

    @Nullable
    @Override
    public Object newP() {
        return null;
    }
}

File: app/src/main/java/cn/peterzhen/demo/MainActivity.kt
package cn.peterzhen.demo

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import butterknife.BindView
import cn.cbsd.mvplibrary.mvp.XActivity
import cn.cbsd.mvplibrary.widget.IosDialog
import cn.cbsd.mvplibrary.widget.OnSheetItemClickListener
import com.afollestad.materialdialogs.MaterialDialog
import com.orhanobut.logger.Logger
import es.dmoral.toasty.MyToast
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch
import timber.log.Timber
import java.util.*

class MainActivity : XActivity() {
    @JvmField
    @BindView(R.id.button1)
    var mButton1: Button? = null

    @JvmField
    @BindView(R.id.button2)
    var mButton2: Button? = null

    @JvmField
    @BindView(R.id.textView)
    var mTextView: TextView? = null

    @JvmField
    @BindView(R.id.activity_main)
    var mActivityMain: LinearLayout? = null
    override val layoutId: Int
        get() = R.layout.activity_main

    override fun initData(savedInstanceState: Bundle?) {
        MyToast.init(application, true, true)
        mButton1!!.setOnClickListener {
            showDialog()
        }
        mButton2!!.setOnClickListener {
            showDialog2()
        }

    }

    private fun showDialog() {
        val text = """
            1.更改：公安修改单位信息不校验必填项；
            2.新增：单位信息和注册信息新增分局审核；
            3.新增：安全检查选择单位可以查看单位详情；
            4.更改：公安管理员账号不能进行安全检查及现场巡查；
            5.更改：辖区统计单位列表字段调整
            1.更改：公安修改单位信息不校验必填项；
            2.新增：单位信息和注册信息新增分局审核；
            3.新增：安全检查选择单位可以查看单位详情；
            4.更改：公安管理员账号不能进行安全检查及现场巡查；
            5.更改：辖区统计单位列表字段调整
            1.更改：公安修改单位信息不校验必填项；
            2.新增：单位信息和注册信息新增分局审核；
            3.新增：安全检查选择单位可以查看单位详情；
            4.更改：公安管理员账号不能进行安全检查及现场巡查；
            5.更改：辖区统计单位列表字段调整
            1.更改：公安修改单位信息不校验必填项；
            2.新增：单位信息和注册信息新增分局审核；
            3.新增：安全检查选择单位可以查看单位详情；
            4.更改：公安管理员账号不能进行安全检查及现场巡查；
            end.更改：辖区统计单位列表字段调整
        """.trimIndent()

        val longText = "是否删除该视频的缴费订单？可以再次登记"

        val list: MutableList<String> = ArrayList()
        for (i in 0..19) {
            list.add("item" + (i + 1))
        }

        IosDialog(context).builder()
                //.setTitle("温馨提示")
                .setMessage(longText)
                //.setItems(list, listener = OnSheetItemClickListener {
                //    getvDelegate().showInfo(String.format("点击的是%d", it))
                //})
                .setPositiveButton("确定", null)
                .setNegativeButton("取消", null)
                .show()
    }

    private fun showDialog2() {
        var page =1
        do {
            Timber.d(page.toString())
            page++
        }while (page<10)
        //val list: MutableList<String?> = ArrayList()
        //for (i in 0..19) {
        //    list.add("item" + (i + 1))
        //}
        //MaterialDialog.Builder(context)
        //        .items(list)
        //        .itemsCallbackSingleChoice(0) { dialog: MaterialDialog?, itemView: View?, which: Int, text: CharSequence? ->
        //            getvDelegate().showInfo(String.format("点击的是%d", which))
        //            false
        //        }
        //        .show()

        //AlertDialog.Builder(context)
        //        .setTitle("温馨提示")
        //        .setMessage("公安修改单位信息不校验必填项")
        //        .setPositiveButton("确定", null)
        //        .setNegativeButton("取消", null)
        //        .show()
    }

    override fun newP(): Any? {
        return null
    }
}